packages/runner-shared/src/event-manager.js

```diff
@@ -313,6 +321,12 @@ export const eventManager = {
       this._clearAllCookies()
       this._setUnload()
     })
+
+    // The window.top should not change between test reloads, and we only need to bind the message event once
+    // Forward all message events to the current instance of the multi-origin communicator
+    window.top?.addEventListener('message', ({ data, source }) => {
```

Comment:
weird that `window.top` can even be `null` here. I guess according to the spec it can be `null` if the `<iframe>` has been discarded: https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc

is there a reason we'd want Cypress to work... without window.top? in `event-manager`, surely we have access to window.top? 

if there's no case for it, then might as well assert to make this stricter, otherwise I can see this unintentionally causing problems:

```suggestion
    if (!window.top) throw new Error('missing window.top in event-manager')

    window.top.addEventListener('message', ({ data, source }) => {
```

Reply:
@mjhenkes Any thoughts on this? Looks like you added the optional chaining to `top` in https://github.com/cypress-io/cypress/pull/20354

Reply:
I added the optional chaining to satisfy typescript, I think @flotwig  proposal is a a better solution.